Runtime/Scripts/PersistentConnection.cs

```diff
@@ -131,192 +217,193 @@ IEnumerator OnIntervalEdgeEventsLocation(EdgeEventsConnection connection)
+      yield return StartCoroutine(LocationService.EnsureLocation());
+      if (hasTCPPorts)
+      {
+        connection.TestConnectAndPostLatencyUpdate(host, (uint)config.latencyTestPort, location).ConfigureAwait(false);
+      }
+      else
+      {
+        integration.matchingEngine.EdgeEventsConnection
+          .TestPingAndPostLatencyUpdate
+          (integration.GetHost(), location).ConfigureAwait(false);
+      }
+      latencyUpdatesCounter++;
+      yield return StartCoroutine(OnIntervalEdgeEventsLatency(connection, host));
+    }
+
+    async void HandleReceivedEvents(ServerEdgeEvent edgeEvent)
+    {
+      Logger.Log("Received event type: " + edgeEvent.EventType);
+      FindCloudletEventTrigger trigger = event_trigger_dict[edgeEvent.EventType];
+      if (config.newFindCloudletEventTriggers.Contains(trigger))
+      {
+        switch (trigger)
+        {
+          case FindCloudletEventTrigger.Error:
+            PropagateError(FindCloudletEventTrigger.Error, edgeEvent.ErrorMsg);
+            return;
+          case FindCloudletEventTrigger.LatencyTooHigh:
+            ProcessLatency(edgeEvent.Statistics);
+            break;
+          default:
+            if (edgeEvent.NewCloudlet != null)
             {
-              latencyUpdatesRunning = false;
-              yield break;
+              PropagateSuccess(trigger, edgeEvent.NewCloudlet);
             }
-          }
-          yield return new WaitForSecondsRealtime(config.latencyConfig.updateIntervalSeconds);
-          yield return StartCoroutine(LocationService.EnsureLocation());
+            return;
+        }
+      }
+      else
+      {
+        if (edgeEvent.EventType == ServerEventType.EventLatencyRequest)
+        {
+          string host = integration.GetHost();
+          Loc location = LocationService.RetrieveLocation();
           if (hasTCPPorts)
           {
-            connection.TestConnectAndPostLatencyUpdate(host, (uint)config.latencyTestPort, location).ConfigureAwait(false);
+            await integration.matchingEngine.EdgeEventsConnection.TestConnectAndPostLatencyUpdate(host, (uint)config.latencyTestPort, location);
           }
           else
           {
-            integration.matchingEngine.EdgeEventsConnection
-              .TestPingAndPostLatencyUpdate
-              (integration.GetHost(), location).ConfigureAwait(false);
+            await integration.matchingEngine.EdgeEventsConnection.TestPingAndPostLatencyUpdate(host, location);
           }
-          latencyUpdatesCounter++;
-          yield return StartCoroutine(OnIntervalEdgeEventsLatency(connection, host));
         }
+      }
+    }
 
-    async void HandleServerReceivedEvents(ServerEdgeEvent edgeEvent)
+    async void ProcessLatency(Statistics stats)
+    {
+      FindCloudletReply currentCloudlet = integration.latestFindCloudletReply;
+      if (stats.Avg < config.latencyThresholdTriggerMs)
+      {
+        integration.UseFindCloudletPerformanceMode(true);
```

Comment:
hmm. We should reset the FindCloudletPerformanceMode after the FIndCloudlet finishes:

```
prevPerformanceMode = integration.PerformanceMode;
integration.UseFindCloudletPerformanceMode(true);
bool fcResult = await integration.FindCloudlet();
intergration.UseFindCloudletPerformanceMode(prevPerformanceMode);
```

Reply:
I'm actually curious if we really want it a boolean, instead of the Mode, which is a public state Enum.

Reply:
If you are talking about UseFindCloudletPerformanceMode(true)

For setting FindCloudletMode we have a Property with a getter (Mode) and a private variable (mode) where we change the mode internally.

From previous conversations, I think we agreed to expose UseFindCloudletPerformanceMode(true) as the developer way for using PerformanceMode aligning with UseWifiOnly(boolean).

I am just going to change the variable (mode) encapsulation from private to internal so I can access its value it from PresistentConnection.cs

<img width="866" alt="Screen Shot 2021-06-21 at 12 56 39 PM" src="https://user-images.githubusercontent.com/24863504/122820140-21418e00-d290-11eb-9593-021bf126884c.png">

Reply:
has this been fixed?

Reply:
Yes, it have been resolved as mentioned in my comment. Let me know if this is sufficient.

Reply:
i think you should set `integration.mode = prevMode;` right after the FindCloudlet call (ie. line 362)